pytorch_lightning/utilities/data.py

```diff
@@ -208,8 +210,14 @@ def _get_dataloader_init_kwargs(
     if not isinstance(dataloader, DataLoader):
         raise ValueError(f"The dataloader {dataloader} needs to subclass `torch.utils.data.DataLoader`")
 
-    # get the dataloader instance attributes
-    attrs = {k: v for k, v in vars(dataloader).items() if not k.startswith("_")}
+    was_wrapped = hasattr(dataloader, "_set_arg_names")
+    if was_wrapped:
+        attrs = {k: getattr(dataloader, "__" + k) for k in dataloader._set_arg_names}
+        original_dataset = dataloader.__dataset  # we have this saved from _wrap_init
+    else:
+        # get the dataloader instance attributes
+        attrs = {k: v for k, v in vars(dataloader).items() if not k.startswith("_")}
+        original_dataset = attrs["dataset"]
```

Comment:
If it wasn't wrapped, can we do this? The DataLoader might not define it.

Reply:
Well, sort of. `DataLoader` (the PyTorch one) does it and the class has to inherit from it. I can change it to `None`. Again, the only place where this is needed is to figure out the code path on this line https://github.com/PyTorchLightning/pytorch-lightning/pull/12981/files#diff-f21f92490e6c1417367a48324385cf66c908d35fb4321a4b1b20e5ba410830f4R251